MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```diff
@@ -0,0 +1,213 @@
+                }
+                else
+                {
+                    int colonIndex = metaDataRow.IndexOf(':', StringComparison.OrdinalIgnoreCase);
+                    metaDataFieldName = metaDataRow[..colonIndex][1..].Trim();
+                    metaDataFieldValue = metaDataRow[(colonIndex + 1)..][..^1].Trim();
+                }
+
+                fileMetaData.Add(metaDataFieldName, metaDataFieldValue);
+            }
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Error parsing lyric data from {Provider}", Name);
+        }
+
+        if (sortedLyricData.Count == 0)
+        {
+            return null;
+        }
+
+        for (int i = 0; i < sortedLyricData.Count; i++)
+        {
+            var timeData = sortedLyricData[i].TimeTags.First().Value;
+            if (timeData is null)
+            {
+                continue;
+            }
+
+            long ticks = TimeSpan.FromMilliseconds(timeData.Value).Ticks;
+            lyricList.Add(new LyricLine(sortedLyricData[i].Text, ticks));
+        }
+
+        if (fileMetaData.Count != 0)
+        {
+            // Map metaData values from LRC file to LyricMetadata properties
+            LyricMetadata lyricMetadata = MapMetadataValues(fileMetaData);
+
+            return new LyricResponse { Metadata = lyricMetadata, Lyrics = lyricList };
+        }
+
+        return new LyricResponse { Lyrics = lyricList };
+    }
+
+    /// <summary>
+    /// Converts metadata from an LRC file to LyricMetadata properties.
+    /// </summary>
+    /// <param name="metaData">The metadata from the LRC file.</param>
+    /// <returns>A lyricMetadata object with mapped property data.</returns>
+    private LyricMetadata MapMetadataValues(IDictionary<string, string> metaData)
+    {
+        LyricMetadata lyricMetadata = new LyricMetadata();
+
+        if (metaData.TryGetValue("ar", out var artist) && !string.IsNullOrEmpty(artist))
+        {
+            lyricMetadata.Artist = artist;
+        }
+
+        if (metaData.TryGetValue("al", out var album) && !string.IsNullOrEmpty(album))
+        {
+            lyricMetadata.Album = album;
+        }
+
+        if (metaData.TryGetValue("ti", out var title) && !string.IsNullOrEmpty(title))
+        {
+            lyricMetadata.Title = title;
+        }
+
+        if (metaData.TryGetValue("au", out var author) && !string.IsNullOrEmpty(author))
+        {
+            lyricMetadata.Author = author;
+        }
+
+        if (metaData.TryGetValue("length", out var length) && !string.IsNullOrEmpty(length))
+        {
+            // Ensure minutes include leading zero
+            var lengthData = length.Split(':');
+            if (lengthData[0].Length == 1)
+            {
+                length = "0" + length;
+            }
+
+            // If only Minutes and Seconds were provided, prepend zeros for hours
+            if (lengthData.Length == 2)
```

Comment:
Is all this necessary? There's an overload for `DateTime.TryParseExact` that takes in a list of possible formats.